packages/datatrak-web/src/views/AccountSettingsPage/PersonalDetailsSection.tsx

```diff
@@ -4,10 +4,168 @@
+  margin: 0; // Use gap on parent to control spacing
+`;
+
+const PersonalDetailsForm = styled.form`
+  display: grid;
+  gap: 1.56rem 1.25rem;
+  max-width: 44.25rem;
+  width: 100%;
+  ${({ theme }) => theme.breakpoints.up('sm')} {
+    grid-template-columns: repeat(2, 1fr);
+  }
+
+  .MuiFormLabel-root {
+    color: ${props => props.theme.palette.text.primary};
+    font-weight: ${({ theme }) => theme.typography.fontWeightMedium};
+  }
+`;
 
 export const PersonalDetailsSection = () => {
+  const { mutate: updateUser } = useEditUser(() => successToast('Personal details updated'));
+  const user = useCurrentUser();
+
+  const {
+    control,
+    formState: { isDirty, dirtyFields },
+    handleSubmit,
+    register,
+  } = useForm<PersonalDetailsFormFields>({
+    defaultValues: {
+      firstName: user.firstName ?? '',
+      lastName: user.lastName ?? '',
+      mobileNumber: user.mobileNumber ?? '',
+      employer: user.employer ?? '',
+      position: user.position ?? '',
+    },
+  });
+
+  function onSubmit(userDetails): SubmitHandler<PersonalDetailsFormFields> {
+    const updates = Object.keys(dirtyFields)
+      .filter(field => dirtyFields[field])
+      .reduce(
+        (updatedFields, field) => ({ ...updatedFields, [field]: userDetails[field] }),
+        {} as UserAccountDetails,
+      );
+
+    updateUser(updates);
+  }
+
   return (
-    <AccountSettingsSection title="Personal details" description="Edit your personal details" />
+    <AccountSettingsSection title="Personal details" description="Edit your personal details">
+      <PersonalDetailsForm onSubmit={handleSubmit(onSubmit)}>
+        <Controller
+          control={control}
+          name="firstName"
+          as={
+            <StyledTextField
+              label="First name"
+              placeholder="First name"
+              autoComplete="given-name"
+              inputProps={{ enterKeyHint: 'next' }}
+              {...register('firstName', { required: true })}
+              required
+            />
+          }
+        />
+        <Controller
+          control={control}
+          name="lastName"
+          as={
+            <StyledTextField
+              label="Last name"
+              placeholder="Last name"
+              autoComplete="family-name"
+              inputProps={{ enterKeyHint: 'next' }}
+              {...register('lastName', { required: true })}
+              required
+            />
+          }
+        />
+        <StyledTextField
+          name="email"
+          label="Email"
+          placeholder="Email"
+          tooltip="You cannot change your email address"
+          autoComplete="email"
+          value={user.email}
+          required
+          disabled
+        />
+        <Controller
+          control={control}
+          name="mobileNumber"
+          as={
+            <StyledTextField
+              label="Contact number (optional)"
+              placeholder="Contact number"
+              autoComplete="tel"
```

Comment:
```suggestion
              autoComplete="tel"
              type="tel"
```
I think the field type i the most important attribute on the inputs as it will be used for built in validation as well as showing the correct ui such as a number picker on tablet and mobiles for number fields. We usually don't bother about setting the autocomplete names explicitly as it's not supported equally across browsers but there's no harm in including it if you want to.

Reply:
Oops, I had it in my head I’d done this. Thanks for catching it!